### ISSUE TITLE
PurpleAir Layer 

### ISSUE THREAD

**Issue by sagarpreet-chadha** [publiclab/leaflet-environmental-layers#14]
Part of #1 . 

I was reading their API and trying to understand some technical terms (AQI , PM2.5 , etc.) . 
Link to API is : https://www.purpleair.com/json?fetchData=true&minimize=true&sensorsActive2=10080&orderby=L

I guess we want to show : 

- Label (Name of place)

- PM2.5 value

- Temperature

- Humidity

- Pressure

Then Hardware information that we can show is :  

- Type

- Hardware

 

- Version

 

- Uptime

- RSSI

**Comment by sagarpreet-chadha** [comment]
@jywarren ...as PurpleAir API is dynamic (real-time) and the value of API parameters like **PM2.5** keeps changing , but not drastically ! 
So i was thinking of parsing the API only once and not making additional listener to hit the API after every x unit of time to change the value because the values are not changing drastically , and also because it would make the map's movement slow 🐌 .

What do you think ?

**Comment by sagarpreet-chadha** [comment]
If you will check the map: 
 https://www.purpleair.com/map?&zoom=3&detailsshowing=7522&selected=145558|145560|427956|427958&lat=37.702223598035076&lng=8.537079834915971&clustersize=41&orderby=L&latr=64.33028522187323&lngr=253.125

It is parsing 3908 records at a time , right ? 
So what would be the best way of parsing data without making the map laggy as there seems to be no provision of putting coordinates in the API .

**Comment by jywarren** [comment]
Oh wow... very impressive!

Hmm. Let's think about a static version first, and put the accessed timestamp somewhere visible. Then let's open a new issue to think about a dynamic version.

I'm starting to think that some centralized caching mechanism may be an important thing for load time if lots of people use this map at once, esp. for the less stable APIs like the government ones. 

Is there a way to request heat map tiles for the Purple Air map, in a way we can display with less work, but offer to display specific stations if people zoom in far enough?

**Comment by jywarren** [comment]
Also, just to say, this is very very impressive @sagarpreet-chadha !! 

Some of these questions like the AQI are appropriate for PublicLab.org as lots of people there probably have opinions!

**Comment by sagarpreet-chadha** [comment]
I have got response on my question on publiclab.org , it seems it is better to use PM2.5 value .
Submitting a patch for the same .

**Comment by sagarpreet-chadha** [comment]
PR #15 AND #19 .

**Comment by sagarpreet-chadha** [comment]
Merged #15 and #19 .
